src/Analyzers/MSTest.Analyzers.CodeFixes/AssertionArgsShouldBePassedInCorrectOrderFixer.cs

```diff
@@ -0,0 +1,89 @@
+[Shared]
+public sealed class AssertionArgsShouldBePassedInCorrectOrderFixer : CodeFixProvider
+{
+    public sealed override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = ImmutableArray.Create(DiagnosticIds.AssertionArgsShouldBePassedInCorrectOrderRuleId);
+
+    public override FixAllProvider GetFixAllProvider()
+        // See https://github.com/dotnet/roslyn/blob/main/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+        => WellKnownFixAllProviders.BatchFixer;
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        Diagnostic diagnostic = context.Diagnostics[0];
+        TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+
+        SyntaxNode root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+        if (root.FindNode(diagnosticSpan) is not InvocationExpressionSyntax invocationExpr)
+        {
+            return;
+        }
+
+        if (context.Diagnostics.Any(d => !d.Properties.ContainsKey(DiagnosticDescriptorHelper.CannotFixPropertyKey)))
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    CodeFixResources.FixAssertionArgsOrder,
+                    ct => SwapArgumentsAsync(context.Document, invocationExpr, ct),
+                    nameof(AssertionArgsShouldBePassedInCorrectOrderFixer)),
+                context.Diagnostics);
+        }
+    }
+
+    private static async Task<Document> SwapArgumentsAsync(Document document, InvocationExpressionSyntax invocationExpr, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+
+        ArgumentListSyntax argumentList = invocationExpr.ArgumentList;
+        SeparatedSyntaxList<ArgumentSyntax> arguments = argumentList.Arguments;
+
+        ArgumentSyntax expectedArg = arguments.FirstOrDefault(arg => IsExpectedArgument(arg));
+        ArgumentSyntax actualArg = arguments.FirstOrDefault(arg => IsActualArgument(arg));
+
+        // Handle positional arguments if named arguments are not found
+        if (expectedArg == null || actualArg == null)
+        {
+            expectedArg = arguments[0];
+            actualArg = arguments[1];
+        }
+
+        var newArguments = arguments.ToList();
+        int expectedIndex = arguments.IndexOf(expectedArg);
+        int actualIndex = arguments.IndexOf(actualArg);
+
+        ArgumentSyntax tmpExpectedArg = expectedArg;
+        newArguments[expectedIndex] = expectedArg.WithExpression(actualArg.Expression);
+        newArguments[actualIndex] = actualArg.WithExpression(tmpExpectedArg.Expression);
+
+        InvocationExpressionSyntax newInvocationExpr = invocationExpr.WithArgumentList(SyntaxFactory.ArgumentList(SyntaxFactory.SeparatedList(newArguments)));
+        SyntaxNode root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
It's already computed on the lambda outside, it'd be better to just flow it to this method.